src/post.ts

```diff
@@ -149,12 +149,18 @@ function juneCleave(): void {
 export default function postCombatActions(skipDiet = false): void {
   juneCleave();
   numberology();
-  if (!skipDiet && !globalOptions.noDiet) {
+
+  skipDiet ||= globalOptions.noDiet;
+
+  if (!skipDiet) {
     fillPantsgivingFullness();
     fillSweatyLiver();
   }
   coldMedicineCabinet();
   safeInterrupt();
   safeRestore();
   updateMallPrices();
+  if (myAdventures() === 0 && globalOptions.nightcap && globalOptions.overdrunk && !skipDiet) {
+    runNightcap();
```

Comment:
I would prefer a structure where nightcap gets run only in one spot, rather than being in two separate places for two separate scenarios.

I am by no means married to this. I'm very much a bachelor in this situation.

Reply:
I will consider this. It comes down into the cleanest way to organize the code, and this (at hte time) seemed easier than reworking bigger parts of the adventuring code

Reply:
Supes reasonable. I'm going to get one more pair of eyes on this maybe and then call it a day